crates/ruff_linter/src/rules/flake8_pie/rules/unnecessary_spread.rs

```diff
@@ -45,7 +50,22 @@ pub(crate) fn unnecessary_spread(checker: &mut Checker, keys: &[Option<Expr>], v
             // We only care about when the key is None which indicates a spread `**`
             // inside a dict.
             if let Expr::Dict(_) = value {
-                let diagnostic = Diagnostic::new(UnnecessarySpread, value.range());
+                let mut diagnostic = Diagnostic::new(UnnecessarySpread, value.range());
+                if checker.settings.preview.is_enabled() {
+                    let range = value.range();
+                    // unwrap -- `item.0 == None` iff this is a spread operator
+                    // which means there *must* be a `**` here
+                    let doublestar = checker.locator().up_to(range.start()).rfind("**").unwrap();
```

Comment:
I wasn't sure if there was a better way to do this -- the `key` is just `None`, and AFAICT nothing in the AST actually records the location of the `**`, so doing this `rfind` is the best way I could think of (at least w/o changing the AST).

Reply:
You could use our `SimpleTokenizer`, which is basically a stripped-down, zero-allocation lexer that's useful in cases like this. If you grep for `SimpleTokenizer::starts_at`, you can find some examples.